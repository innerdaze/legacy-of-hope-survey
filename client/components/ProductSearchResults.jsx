import React, { Component } from 'react'
import PropTypes from 'prop-types'
import List from 'grommet/components/List'
import ListItem from 'grommet/components/ListItem'

class ProductSearchResults extends Component {
  render() {
    return (
      <List>
        {this.props.items.length &&
          this.props.items.map(item => (
            <ListItem justify='between'
              separator='horizontal'>
              <span>
                {item.ProductName}
              </span>
            </ListItem>
        ))}
      </List>
    )
  }
}

ProductSearchResults.propTypes = {
  results: PropTypes.array.isRequired
}

ProductSearchResults.defaultProps = {
  results: []
}

export default ProductSearchResults
