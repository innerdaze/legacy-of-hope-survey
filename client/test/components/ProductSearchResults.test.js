import React from 'react'
import ProductSearchResults from '../../components/ProductSearchResults'

let wrapper, items

describe('Testing on ProductSearchResults...', () => {
  beforeEach(() => {
    items = [
      {
        ProductID: 'FooID',
        ProductName: 'BarName'
      }
    ]
    wrapper = shallow(<ProductSearchResults items={items} />)
  })

  describe('ProductSearchResults layout', () => {
    test('Expect to have a List and ListItem comps', () => {
      expect(wrapper.find('List')).toHaveLength(1)
      expect(wrapper.find('ListItem')).toHaveLength(1)

      expect(wrapper).toMatchSnapshot()
    })

    test('Expect to have a List and without ListItem comp if items props is empty array', () => {
      items = []
      wrapper = shallow(<ProductSearchResults items={items} />)
      expect(wrapper.find('List')).toHaveLength(1)
      expect(wrapper.find('ListItem')).toHaveLength(0)

      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('Testing on prop', () => {
    test('Expect to have multiple ListItem if more items passed in', () => {
      items = [
        ...items,
        {
          ProductID: 'FooID2',
          ProductName: 'FooName2'
        },
        {
          ProductID: 'FooID3',
          ProductName: 'FooName3'
        }
      ]

      wrapper = shallow(<ProductSearchResults items={items} />)

      expect(wrapper.find('List')).toHaveLength(1)
      expect(wrapper.find('ListItem').length).toBeGreaterThan(1)

      expect(wrapper).toMatchSnapshot()
    })
  })
})
