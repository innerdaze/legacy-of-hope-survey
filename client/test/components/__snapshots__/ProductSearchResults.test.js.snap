// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing on ProductSearchResults... ProductSearchResults layout Expect to have a List and ListItem comps 1`] = `
<List
  role="list"
>
  <ListItem
    align="center"
    direction="row"
    justify="between"
    key="FooID"
    pad={
      Object {
        "horizontal": "medium",
        "vertical": "small",
      }
    }
    role="listitem"
    separator="horizontal"
  >
    <span>
      BarName
    </span>
  </ListItem>
</List>
`;

exports[`Testing on ProductSearchResults... ProductSearchResults layout Expect to have a List and without ListItem comp if items props is empty array 1`] = `
<List
  role="list"
>
  0
</List>
`;

exports[`Testing on ProductSearchResults... Testing on prop Expect to have multiple ListItem if more items passed in 1`] = `
<List
  role="list"
>
  <ListItem
    align="center"
    direction="row"
    justify="between"
    key="FooID"
    pad={
      Object {
        "horizontal": "medium",
        "vertical": "small",
      }
    }
    role="listitem"
    separator="horizontal"
  >
    <span>
      BarName
    </span>
  </ListItem>
  <ListItem
    align="center"
    direction="row"
    justify="between"
    key="FooID2"
    pad={
      Object {
        "horizontal": "medium",
        "vertical": "small",
      }
    }
    role="listitem"
    separator="horizontal"
  >
    <span>
      FooName2
    </span>
  </ListItem>
  <ListItem
    align="center"
    direction="row"
    justify="between"
    key="FooID3"
    pad={
      Object {
        "horizontal": "medium",
        "vertical": "small",
      }
    }
    role="listitem"
    separator="horizontal"
  >
    <span>
      FooName3
    </span>
  </ListItem>
</List>
`;
